ui-tests/cypress/integration/administration/file-browser.spec.js

```diff
@@ -1,92 +1,525 @@
+      currentPage = currentPage.getContent().openCreateTextFilePage();
+      cy.validateAppBuilderUrlPathname('/file-browser/create-text-file');
+      currentPage.getContent().getNameInput().type(textTestFile.name);
+      currentPage.getContent().getTextArea().type(textTestFile.content);
+      currentPage = currentPage.getContent().clickFileBrowserBreadCrumbs(1);
+      cy.validateAppBuilderUrlPathname('/file-browser');
+      currentPage.getContent().getFilesTable().should('exist').and('be.visible');
+      currentPage.getContent().getTableRows().should('have.length', 7);
+      currentPage.getContent().getFolderLink(-1).should('contain', testFolderInfo.name);
+      currentPage.getContent().openSubFolder(-1);
+      currentPage.getContent().getFilesTable().should('not.exist');
+      currentPage.getContent().getEmptyFolderAlert().should('exist');
+    });
+
+    it([Tag.FEATURE, 'ENG-3297'], 'No file is created when canceling out of create text file page', () => {
+      currentPage = openFileBrowserPage();
+      cy.validateAppBuilderUrlPathname('/file-browser');
+      currentPage.getContent().openSubFolder(0);
+      currentPage = currentPage.getContent().openCreateTextFilePage();
+      cy.validateAppBuilderUrlPathname('/file-browser/create-text-file');
+      currentPage.getContent().getNameInput().type(textTestFile.name);
+      currentPage.getContent().getTextArea().type(textTestFile.content);
+      currentPage = currentPage.getContent().clickCancelButton();
+      cy.validateAppBuilderUrlPathname('/file-browser');
+      currentPage.getContent().getFilesTable().should('exist');
+      currentPage.getContent().getTableRows().should('have.length', 6);
+      currentPage.getContent().getFileKebabMenu(`${textTestFile.name}.${textTestFile.extension}`).should('not.exist');
     });
 
   });
 
+  const textTestFile = {name: 'test', extension: 'txt', content: 'this is a test'}
+
+  const testFileInfo          = {path: '',      name: 'data1.json', base64: '', type: 'application/json'}
+  const subfolderTestFileInfo = {path: 'test/', name: 'data1.json', base64: '', type: 'application/json'}
+
+  const createTestFile = (fileInfo) => {
+    cy.fileBrowserController().then(controller => controller.createFile(fileInfo, false))
+      .then(response => {
+        if (response.status == 200) cy.wrap([fileInfo.path + fileInfo.name]).as('filesToBeDeleted')
+      });
+  }
+
+  const testFolderInfo          = {name: 'test',    path: ''}
+  const subfolderTestFolderInfo = {name: 'subTest', path: 'test/'}
+
+  const createTestFolder = (folderInfo) => {
+    cy.fileBrowserController().then(controller => controller.createFolder(folderInfo.path + folderInfo.name))
+                              .then(response => {
+                                cy.get('@folderToBeDeleted').then(folder => {
+                                 if (!folder) cy.wrap(response.body.payload.path).as('folderToBeDeleted')
```

Comment:
how it can be already existing?

Reply:
When creating a subfolder, I made it so that only the parent folder gets deleted at the end, which also deletes the subfolder. If I didn't put this, then the reference to the parent folder would be overwritten by the reference to the subfolder, and only the subfolder would get deleted instead